File: src/test/java/club/geek66/querydsl/OperatorUtilsTest.java
package club.geek66.querydsl;

import com.querydsl.core.types.Ops;
import io.vavr.control.Either;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * @author: orange
 * @date: 2020/12/3
 * @time: 下午4:13
 * @copyright: Copyright 2020 by orange
 */
class OperatorUtilsTest {

	@Test
	public void testParseOps() {
		Either<OperatorUtils.ParseOpsError, Ops> gtResult = OperatorUtils.parseOps(">");
		Assertions.assertEquals(Either.right(Ops.GT), gtResult);

		Either<OperatorUtils.ParseOpsError, Ops> ltResult = OperatorUtils.parseOps("<");
		Assertions.assertEquals(Either.right(Ops.LT), ltResult);

		Either<OperatorUtils.ParseOpsError, Ops> goeResult = OperatorUtils.parseOps(">=");
		Assertions.assertEquals(Either.right(Ops.GOE), goeResult);

		Either<OperatorUtils.ParseOpsError, Ops> loeResult = OperatorUtils.parseOps("<=");
		Assertions.assertEquals(Either.right(Ops.LOE), loeResult);
	}

}

File: src/test/kotlin/club/geek66/querydsl/OperatorUtilsTest.kt
package club.geek66.querydsl

import arrow.core.Either
import club.geek66.querydsl.OperatorUtils.parseOps
import com.querydsl.core.types.Ops
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/**
 * @author: orange
 * @date: 2020/12/3
 * @time: 下午4:13
 * @copyright: Copyright 2020 by orange
 */
internal class OperatorUtilsTest {

	@Test
	fun testParseOps() {
		val gtResult = parseOps(">")
		Assertions.assertEquals(Either.right(Ops.GT), gtResult)
		val ltResult = parseOps("<")
		Assertions.assertEquals(Either.right(Ops.LT), ltResult)
		val goeResult = parseOps(">=")
		Assertions.assertEquals(Either.right(Ops.GOE), goeResult)
		val loeResult = parseOps("<=")
		Assertions.assertEquals(Either.right(Ops.LOE), loeResult)
	}
}